internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -114,6 +114,12 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 			pluginsdk.ForceNewIfChange("network_profile.0.network_policy", func(ctx context.Context, old, new, meta interface{}) bool {
 				return old.(string) != "" || new.(string) != string(managedclusters.NetworkPolicyCilium)
 			}),
+			pluginsdk.ForceNewIfChange("network_profile.0.pod_cidr", func(ctx context.Context, old, new, meta interface{}) bool {
+				return old.(string) != ""
+			}),
+			pluginsdk.ForceNewIfChange("network_profile.0.pod_cidrs", func(ctx context.Context, old, new, meta interface{}) bool {
+				return len(old.([]interface{})) > 0
+			}),
```

Comment:
Would you be able to check whether `pod_cidr` and `pod_cidrs` can be updated after a value has been set for them? If the API allows updates for this property then we won't need to add this to the `CustomizeDiff`.

Reply:
The API would accept new values for pod_cidr and pod_cidrs, but ignores the new values und report Succeeded without any changed.

Ref: https://github.com/Azure/azure-cli/issues/28403 (and I test that locally)